src/tile_setter.rs

```diff
@@ -0,0 +1,213 @@
+    /// may still inform the vector that there is space for a few more elements.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    /// use bevy::prelude::*;
+    ///
+    /// let mut setter = TileSetter::with_capacity(10);
+    /// let coord = Vec3::new(1., 1., 0.);
+    /// let tile = Tile::new(1);
+    /// setter.push(coord, tile, 0);
+    /// assert_eq!(setter.capacity(), 10);
+    /// setter.shrink_to_fit();
+    /// assert!(setter.capacity() >= 1);
+    /// ```
+    pub fn shrink_to_fit(&mut self) {
+        self.0.shrink_to_fit();
+    }
+
+    /// Reserves the minimum capacity for exactly additional more elements to be
+    /// inserted in the given `TileSetter`.
+    ///
+    /// After calling `reserve_exact`, capacity will be greater than or equal to
+    /// self.len() + additional. Does nothing if the capacity is already
+    /// sufficient.
+    ///
+    /// Note that the allocator may give the collection more space than it
+    /// requests. Therefore, capacity can not be relied upon to be precisely
+    /// normal. Prefer `reserve` if future insertions are expected.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the new capacity overflows `usize`.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    ///
+    /// let mut setter = TileSetter::new();
+    /// setter.reserve_exact(10);
+    /// assert!(setter.capacity() >= 10);
+    /// ```
+    pub fn reserve_exact(&mut self, additional: usize) {
+        self.0.reserve_exact(additional);
+    }
+
+    /// Returns the number of elements in the `TileSetter`, also referred to as
+    /// its 'length'.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    ///
+    /// let mut setter = TileSetter::new();
+    /// assert_eq!(setter.len(), 0);
+    /// ```
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Returns the number of elements in the vector can hold without
+    /// reallocating.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    ///
+    /// let mut setter = TileSetter::with_capacity(10);
+    /// assert_eq!(setter.capacity(), 10);
+    /// ```
+    pub fn capacity(&self) -> usize {
+        self.0.capacity()
+    }
+
+    /// Returns if the `TileSetter` is empty and contains no elements.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_tilemap::prelude::*;
+    /// use bevy::prelude::*;
+    ///
+    /// let mut setter = TileSetter::new();
+    /// assert!(setter.is_empty());
+    ///
+    /// let coord = Vec3::new(1., 1., 0.);
+    /// let tile = Tile::new(1);
+    /// setter.push(coord, tile, 0);
+    /// assert!(!setter.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Returns an iterator over all coordinates and tiles in the `TileSetter`.
+    pub fn iter(&self) -> Iter<'_, (Vec3, Tile, usize)> {
```

Comment:
That is a good point.

Reply:
Letting Vec be the thing the user uses for complex tile setups would also address my comments on shrink_to_fi and reserve_exact because they wouldn't be necessary either even for the power user case.

Reply:
Resolved now as TileSetter is deprected.